plugins/statistics.koplugin/main.lua

```diff
@@ -175,14 +200,44 @@ function ReaderStatistics:initData()
     -- Update these numbers to what's actually stored in the settings
     -- (not that "notes" is invalid and does not represent edited highlights)
     self.data.highlights, self.data.notes = self.ui.bookmark:getNumberOfHighlightsAndNotes()
-    self.curr_total_time = 0
-    self.curr_total_pages = 0
     self.id_curr_book = self:getIdBookDB()
-    self.total_read_pages, self.total_read_time = self:getPageTimeTotalStats(self.id_curr_book)
-    if self.total_read_pages > 0 then
-        self.avg_time = self.total_read_time / self.total_read_pages
+    self.book_read_pages, self.book_read_time = self:getPageTimeTotalStats(self.id_curr_book)
+    if self.book_read_pages > 0 then
+        self.avg_time = self.book_read_time / self.book_read_pages
     else
-        self.avg_time = 0
+        -- NOTE: Possibly less weird-looking than initializing this to 0?
+        self.avg_time = math.floor(0.50 * self.page_max_read_sec)
+        logger.info("ReaderStatistics: Initializing average time per page at 50% of the max value, i.e.,", self.avg_time)
+    end
+end
+
+-- Reset the (volatile) stats on page count changes (e.g., after a font size update)
+function ReaderStatistics:onUpdateToc()
+    local new_pagecount = self.view.document:getPageCount()
+
+    if new_pagecount ~= self.data.pages then
+        logger.info("ReaderStatistics: Pagecount change, flushing volatile book statistics")
```

Comment:
Do these 2 needs to be logger.info() ? I was surprised to see them in my log on my Kobo and wondered what to make out of that info :)

Reply:
I think there may be a few other instances of those in there that might be better classified as something other than info, yeah.

Certainly, the second one here. I could argue that the first one is mildly informative, but then again the fact that it'll show on every new book makes it fairly verbose.